### PR TITLE
Add benchmark for scheduling of pods with PVs

### PR DESCRIPTION
**Pull request by jsafrane**
/kind feature
Part of https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190408-volume-scheduling-limits.md

**What this PR does / why we need it**:
As part of aforementioned KEP, we need to extend current scheduler benchmark to test pods with PVs and their current volume limit predicates (`MaxCSIVolumeCountPred`, `MaxEBSVolumeCount`, `MaxGCEPDVolumeCount`, `MaxAzureDiskVolumeCount`, `MaxCinderVolumeCount`).

This PR adds benchmark for Secrets, CSI and AWS EBS volumes. All the other in-tree cloud predicates share the same code, so only AWS is tested. Secrets are used to see how efficiently the predicates skip unrelated volumes.

The test suite was enhanced to create bound PVs and PVCs and add CSI volume limits to `node.status.allocatable` and to create CSINode objects for each node. Existing framework did not provide any facility to create CSINode when Node is created, I extended `PrepareNodeStrategy` with custom hooks.

```release-note
NONE
```

@bertinatto @gnufied @msau42 @bsalamat, PTAL and comment on the structure.

### CODE REVIEW

File: test/integration/scheduler_perf/scheduler_bench_test.go [hunk 1]
```diff
@@ -79,6 +85,88 @@ func BenchmarkSchedulingPodAntiAffinity(b *testing.B) {
...
+	// The test strategy creates pods with CSI volume via PV.
+	baseClaim := makeBasePersistentVolumeClaim()
+	basePod := makeBasePod()
+	testStrategy := testutils.NewCreatePodWithPersistentVolumeStrategy(baseClaim, csiVolumeFactory, basePod)
+
+	// Each node can use the same amount of CSI volumes as in-tree AWS volume
+	// plugin, so the results should be comparable with BenchmarkSchedulingInTreePVs.
+	driverKey := util.GetCSIAttachLimitKey(testCSIDriver)
+	allocatable := map[v1.ResourceName]string{
+		v1.ResourceName(driverKey): fmt.Sprintf("%d", util.DefaultMaxEBSVolumes),
```
Comment (msau42) [inline test/integration/scheduler_perf/scheduler_bench_test.go]: Can we make the limit used in the test configurable?

**Comment by jsafrane** [comment]
I added CSINode creation. Now I'm waiting for the scheduler changes in #77595
to be able to actually measure something.

**Comment by jsafrane** [comment]
@kubernetes/sig-scalability-pr-reviews, PTAL. I extended `test/utils/runners.go` with optional creation of CSINode instance for each node and PV+PVC for each pod. Please check it goes the right direction or suggest better approach.

I'm still waiting for API changes and updated predicates for the KEP linked above (see https://github.com/kubernetes/kubernetes/pull/77595), but I consider the benchmark framework complete and ready for review.

File: test/integration/scheduler_perf/scheduler_bench_test.go (hunk 1 above)
Comment (jsafrane) [reply test/integration/scheduler_perf/scheduler_bench_test.go]: IMO it's quite useless. In the most 'dense' test, each node gets 10 pods.

**Comment by jsafrane** [comment]
#77595 got merged, so I removed WIP. Ready for full review and merge!

**Comment by jsafrane** [comment]
> This PR may require API review.

This PR does not change any API. All changes are in `test/` directory.

File: test/integration/scheduler_perf/scheduler_bench_test.go (hunk 1 above)
Comment (msau42) [reply test/integration/scheduler_perf/scheduler_bench_test.go]: Interesting why we don't try to max out the 100 pods/node limit in the benchmark.  I would prefer if we can test our "worst case" limits, ie ~100 volumes per node.

**Comment by msau42** [comment]
/lgtm
/assign @bsalamat 
for scheduler review

**Review by bsalamat — COMMENTED** [review]
/lgtm

I don't have approval power though.

**Pull request merged**